src/Toolkit/Toolkit/UI/Controls/TimeSlider/TimeSlider.cs

```diff
@@ -857,6 +857,12 @@ private void CalculateTimeSteps()
 
             for (var nextStep = startTime.AddTimeValue(timeStep); nextStep <= endTime; nextStep = nextStep.AddTimeValue(timeStep))
             {
+                // TODO = come up with a more rigorous definition or expose API
+                if (endTime.Subtract(nextStep).TotalMinutes < 10)
+                {
+                    nextStep = endTime;
+                }
```

Comment:
The intent is to snap the end of the last interval to the end of the time extent in cases where imprecision in time manipulation leads to the combined length of time steps being slightly loo long. 

e.g. 10years/6 intervals = 1.67 year intervals; 1.67 years*6 intervals = 10.2 years; previously in cases like that, the last time interval would just be skipped.

Other changes in TimeExtensions are intended to make the time manipulation as precise as possible (milliseconds). So ideally any error is limited to milliseconds rather than years. This seems to make things work for some common cases, but it really doesn't solve the bigger problems with the control.